src/streamed/index.ts

```diff
@@ -207,6 +218,23 @@ export class StreamedOrderbook {
     })
     return events as AnyEvent<BatchExchange>[]
   }
+
+  /**
+   * Retrieves the batch ID at a given block number.
+   *
+   * @remarks
+   * The batch ID is locally calculated from the block header timestamp as it is
+   * more reliable than executing an `eth_call` to calculate the batch ID on the
+   * EVM since an archive node is required for sufficiently old blocks.
+   */
+  private async getBatchId(blockNumber: BlockNumber): Promise<number> {
+    const BATCH_DURATION: number = 300
```

Comment:
Nice to move this in here! I guess this was the only place it was ever accessed.